buildpack/telemetry/metrics.py

```diff
@@ -555,7 +606,13 @@ def _select_stats_to_emit(self):
                 self._inject_storage_stats,
                 self._inject_health,
             ]
-        selected_stats.append(self._inject_m2ee_stats)
+
+        if not self.micrometer_metrics_enabled:
+            selected_stats.append(self._inject_m2ee_stats)
+        else:
+            selected_stats.append(self._inject_smap_stats)
+        selected_stats.append(self._inject_critical_log_stats)
+
```

Comment:
this covers smap stats and the critical log stats. Where's the logic for all other metrics?

Reply:
[line 604](https://github.com/mendix/cf-mendix-buildpack/blob/LM-591_switchover_to_micrometer/buildpack/telemetry/metrics.py#L604) preps all the relevant metrics we need via the old stream. 
We need to turn off `inject_m2ee_stats`, but that would disable `critical_logs_count` and `smaps` related metrics too. That's why I've separated out those two.